File: In Kotlin/Pascal-Triangle.kt
fun fact(n: Int): Int {
    var f = 1
    for (i in 1..n) f *= i
    return f
}

fun ncr(n: Int, r: Int): Int {
    return fact(n) / (fact(r) * fact(n - r))
}

fun main() {
    print("Enter number of rows: ")
    val n = readln().toInt()

    for (i in 0 until n) {
        print(" ".repeat(n - i))
        for (j in 0..i) {
            print("${ncr(i, j)} ")
        }
        println()
    }
}

File: In TypeScript/Pascal-Triangle.ts
function fact(n: number): number {
    let f = 1;
    for (let i = 1; i <= n; i++) f *= i;
    return f;
}

function ncr(n: number, r: number): number {
    return fact(n) / (fact(r) * fact(n - r));
}

const readline = require("readline-sync");
const n = parseInt(readline.question("Enter number of rows: "));

for (let i = 0; i < n; i++) {
    let row = " ".repeat(n - i);
    for (let j = 0; j <= i; j++) {
        row += ncr(i, j) + " ";
    }
    console.log(row);
}
